resin/resin/resin.cpp

```diff
@@ -467,41 +415,313 @@ bool Resin::on_window_resize(WindowResizeEvent& e) {
   return false;
 }
 
-bool Resin::on_test(WindowTestEvent&) {
-  camera_->set_orthographic(!camera_->is_orthographic());
-  shader_->set_uniform("u_ortho", camera_->is_orthographic());
-  shader_->set_uniform("u_camSize", camera_->height());
-
-  return false;
-}
-
-bool Resin::on_click(MouseButtonPressedEvent& e) {
+bool Resin::on_mouse_btn_pressed(MouseButtonPressedEvent& e) {
   ImGuiIO& io = ImGui::GetIO();
   if (io.WantCaptureMouse) {
     return true;
   }
 
+  if (e.button() == mouse::Code::MouseButtonMiddle) {
+    return activate_orbiting_camera(e.pos());
+  }
+
+  if (e.button() == mouse::Code::MouseButtonRight) {
+    if (glm::dot(camera_->transform.local_up(), glm::vec3(0.0F, 1.0F, 0.0F)) < 0.0F) {
+      return start_interpolation();
+    }
+
+    return activate_first_person_camera(e.pos());
+  }
+
   glm::vec2 relative_pos = e.pos() - viewport_pos_;
   if (relative_pos.x < 0 || relative_pos.y < 0 || relative_pos.x > static_cast<float>(framebuffer_->width()) ||
       relative_pos.y > static_cast<float>(framebuffer_->height())) {
     return false;
   }
 
   if (e.button() == mouse::Code::MouseButtonLeft) {
-    return on_left_click(relative_pos);
+    return select_node(relative_pos);
   }
 
   return false;
 }
 
-bool Resin::on_left_click(glm::vec2 relative_pos) {
-  // TODO(SDF-82): Make sure that mouse pick is not performed when camera is moving
+bool Resin::on_mouse_btn_released(MouseButtonReleasedEvent& e) {
+  if (e.button() == mouse::Code::MouseButtonMiddle) {
+    return deactivate_orbiting_camera();
+  }
+
+  if (e.button() == mouse::Code::MouseButtonRight) {
+    return deactivate_first_person_camera();
+  }
+
+  return false;
+}
+
+bool Resin::on_key_pressed(KeyPressedEvent& e) {
+  if (e.key_code() == key::Code::V) {
```

Comment:
I would change this to ***C*** as in ***C**amera*, as V was just some random key I selected for testing
```suggestion
  if (e.key_code() == key::Code::C) {
```